index.js

```diff
@@ -26,10 +26,27 @@ function componentWillReceiveProps(nextProps) {
   }
 }
 
+function componentWillUpdate(nextProps, nextState) {
+  try {
+    var prevProps = this.props;
+    var prevState = this.state;
+    this.props = nextProps;
+    this.state = nextState;
+    this.__reactInternalSnapshot = this.getSnapshotBeforeUpdate(
```

Comment:
This is the kind of thing for which a `WeakMap` would be good for... but can't use `WeakMap` in React :(

Have you tried at least making it non-enumerable, or even maybe also a polyfill-internal Symbol instead of a strong name, or are those also out when supporting 0.14?

Reply:
I don't think that's necessary. React already stores other "__reactInternal" attributes on class components. And relying on a symbol might cause problems for older browsers (supported by older versions of React) that don't support symbols.

Reply:
Would be nice if a future version of React would use a `WeakMap` for their per-instance internal state instead (React 17?) 🤔